src/Apps/InfiniteDiscovery/InfiniteDiscoveryApp.tsx

```diff
@@ -0,0 +1,372 @@
+import { Flex, Button, Text, Separator } from "@artsy/palette"
+import { getMetaphysicsEndpoint } from "System/Relay/getMetaphysicsEndpoint"
+import React, { useEffect } from "react"
+import { uniq, sampleSize, shuffle } from "lodash"
+
+const ES_URL =
+  "https://vpc-artsy-staging-es7-srsqgcck5vzy7uq2eqqhvrm4x4.us-east-1.es.amazonaws.com"
+
+const BUILD_SEARCH_QUERY = (lArtworks, dArtworks, cArtworks) => {
+  const likedArtworks = lArtworks.map(artwork => {
+    return {
+      _index: "artworks_7shards_production",
+      _type: "_doc",
+      _id: artwork.internalID,
+    }
+  })
+
+  const dismissedArtworks = dArtworks.map(artwork => ({
+    term: {
+      _id: artwork.internalID,
+    },
+  }))
+
+  const query = {
+    query: {
+      bool: {
+        should: [
+          {
+            more_like_this: {
+              fields: ["id", "genes_vectors"],
+              like: likedArtworks,
+              min_term_freq: 1,
+              min_doc_freq: 30,
+              max_query_terms: 18,
+              minimum_should_match: "30%",
+            },
+          },
+        ],
+        must: [
+          {
+            term: { published: true },
+          },
+          {
+            term: { for_sale: true },
+          },
+        ],
+        filter: [
+          { range: { career_stage: { gte: 0, lte: 100 } } },
+          {
+            script: {
+              script: {
+                source: "doc['genes.raw'].size() >= 1",
+              },
+            },
+          },
+        ],
+        must_not: [...(dismissedArtworks || [])],
+      },
+    },
+    collapse: {
+      field: "artist_id",
+    },
+    aggregations: {
+      uniq_artists: {
+        terms: {
+          field: "artist_id",
+          size: 1,
```

Comment:
I think on weaviate we allowed multiple works per artist in a batch... wondering whether this should be set at 2/3 to allow for a little more volume, worth testing I guess.

Reply:
It is the product requirement to have 1 artist/artwork per batch
https://www.notion.so/artsy/Infinite-Discovery-Qualitative-QA-12fcab0764a0805c921bfe0146881f9c?pvs=4#13fcab0764a0804b96eae9d310b58c2d